### DISCUSSION TITLE
v1.3.0 - configurable indicators

### DISCUSSION THREAD

**Discussion by bdarcus** [emacs-citar/citar#754] [General]
The big, breaking-ish, change is [configurable indicators](https://github.com/emacs-citar/citar#indicators).

<hr /><em>This discussion was created from the release <a href='https://github.com/emacs-citar/citar/releases/tag/v1.3.0'>v1.3.0 - configurable indicators</a>.</em>

**Comment by gwbrck** [comment]
After updating, and replacing my citar config with the example out of the wiki, I noticed that citar does look different. For example, the spacing between the icons is gone. How do I get it back?

Thank you!

<img width="77" alt="Bildschirm­foto 2023-03-22 um 20 07 40" src="https://user-images.githubusercontent.com/39911339/227012121-c511c65e-1b0b-4ebe-ae36-c76ed0d57715.png">

**Reply by bdarcus** [reply]
What does your citar config look like?

**Reply by gwbrck** [reply]
```elisp
(use-package citar
  :ensure t
  :demand t
  :after init-bibtex
  :config
  (add-to-list 'citar-bibliography main-bib-file)
  (add-to-list 'citar-library-paths (concat gwbrck/roam "pdfs"))
  (defvar citar-indicator-files-icons
    (citar-indicator-create
     :symbol (all-the-icons-faicon
              "file-o"
              :face 'all-the-icons-green
              :v-adjust -0.1)
     :function #'citar-has-files
     :tag "has:files"))
  (defvar citar-indicator-links-icons
    (citar-indicator-create
     :symbol (all-the-icons-octicon
              "link"
              :face 'all-the-icons-orange
              :v-adjust 0.01)
     :function #'citar-has-links
     :tag "has:links"))
  (defvar citar-indicator-notes-icons
    (citar-indicator-create
     :symbol (all-the-icons-material
              "speaker_notes"
              :face 'all-the-icons-blue
              :v-adjust -0.3)
     :function #'citar-has-notes
     :tag "has:notes"))
  (setq citar-indicators
        (list citar-indicator-files-icons
              citar-indicator-links-icons
              citar-indicator-notes-icons))
  :general
  (leader-key-def
    "fb" 'citar-open))
;;  :hook
;;  (LaTeX-mode . citar-capf-setup)
;;  (org-mode . citar-capf-setup))


(use-package citar-org-roam
  :after citar org-roam
  :ensure t
  :custom
  (citar-org-roam-subdir "annot")
  (citar-org-roam-note-title-template "${author editor} (${year}): ${title}")
  :config
  (citar-org-roam-mode))

(use-package citar-embark
  :ensure t
  :after citar embark
  :no-require
  :config (citar-embark-mode))

(use-package org-cite-csl-activate
  :init
  (unless (package-installed-p 'org-cite-csl-activate)
    (package-vc-install "https://github.com/andras-simonyi/org-cite-csl-activate.git"))
  :hook
  (org-mode . org-cite-csl-activate-render-all)
  (org-mode . cursor-sensor-mode))

(use-package oc
  :no-require
  :config
  (add-to-list 'org-cite-global-bibliography main-bib-file)
  :custom
  (org-cite-csl-styles-dir "~/.config/csl/styles")
  (org-cite-csl-locales-dir "~/.config/csl/locales")
  (org-cite-insert-processor 'citar)
  (org-cite-follow-processor 'citar)
  (org-cite-activate-processor 'csl-activate)
  (org-cite-export-processors '((t csl "apa.csl")))
  :general
  (:keymaps 'org-mode-map
            "C-c b" 'org-cite-insert))
```

**Reply by bdarcus** [reply]
Try adding `(setq citar-symbol-separator " ")`.

I assume that's the issue; will look into if there's something I can do on this end to avoid these kind of surprises.

EDIT: hmm ... that's the default value, so it shouldn't be.

I'll dig a little more.

**Reply by gwbrck** [reply]
Yes, indeed, that was not the problem. I even tried this ```(setq citar-symbol-separator "|")```. It has no effect...

Thanks for your help!

**Reply by bdarcus** [reply]
Super weird.

Here's what I see, but I only get that with that variable set to two spaces (with one I get what you get; just added a note on that to the wiki):

![image](https://user-images.githubusercontent.com/1134/227020325-0462704e-3448-4f35-bed3-a9b42dea04fd.png)

And when I add a bar, it shows up.

EDIT: OK, I see something on my minimal init.

**Reply by bdarcus** [reply]
So I'm stumped ATM.

On my Doom install, it works as expected.

On my more minimal, and experimental, config, if I change the value of `citar-symbol-separator`, I see no change in the minibuffer.

BUT (note the bar, which shows it has an effect) ...

```elisp
ELISP> (citar--make-indicator-symbols "has:notes")
#(" | |N|" 1 2
  (display
   (space :align-to 2))
  3 4
  (display
   (space :align-to 4))
  5 6
  (display
   (space :align-to 6)))
```

As in THAT is what I expect, and I can't explain the discrepancy ATM.

EDIT: after restarting:

![image](https://user-images.githubusercontent.com/1134/227028145-c393dedf-aa92-4b8e-b6c9-3cb5f4539e41.png)


Maybe try all the usual things when stuff goes wonky?

Restarting being the most obvious?

And if you still can't get it working, this is probably worth a bug report.

**Reply by gwbrck** [reply]
Thanks for helping me! 
Double spaces or double pipes or double of whatever leads to the desired outcome: That is just one of these separator symbols. Restarting emacs does not change this. 

That being said, `(citar--make-indicator-symbols "has:notes")` always shows the right amount of separators as it is declared in `citar-symbols-seperator`. So one space is one space and one pipe is one pipe and so on. 

I mean, setting `citar-symbols-seperator` to two spaces instead of one space works... But this feels kind of wrong. I'll check later if I can reproduce this with a minimal config...

**Comment by bdarcus** [comment]
> just doubling the space works... But this feels kind of wrong.

What's "this"? Are you basically suggesting the separator should be defined in the indicator spec?

**Reply by gwbrck** [reply]
I meant setting `citar-symbols-seperator` to `"  "` instead of `" "` just to get it working.

**Reply by bdarcus** [reply]
Yeah; I'll look again at the code. Maybe the width isn't getting calculated correctly using the icons now?
